### DISCUSSION TITLE
Updating Image Roles

### DISCUSSION THREAD

**Discussion by freshleafmedia** [area17/twill#2437] [Ideas]
## Summary

Currently there doesn't appear to be a way of updating existing media roles without manually re-associating the media.

For example, if the database contains instances of `MyModel` (see below) and they have `featured_image`s associated but then the `ratio` of the crop is changed the media is unaffected even after running `php artisan twill:refresh-crops '\App\Models\MyModel' featured_image`

```php
class MyModel extends Model
{
    public $mediasParams = [
        'featured_image' => [
            'default' => [
                [
                    'name' => 'default',
                    'ratio' => 16 / 9,
                ],
            ],
        ],
    ];
}
```

## Describe the solution you'd like

Running `twill:refresh-crops` would update the `mediables` table for crops which have been updated rather than just adding new ones and removing defunct ones

**Comment by freshleafmedia** [comment]
@Eloiducrest I'm afraid that doesn't solve the problem. I want to be able to change the configuration of existing crops and have that reflected when rendered

**Comment by ifox** [comment]
The command only adds or remove, it doesn't edit existing crops because it could override manually curated crops by admins in the CMS. So we could add an option to force it, but one way you can already do this would be by renaming the `default` crop, running the command, renaming it back to `default`, and finally running the command again. I guess this is not ideal in a scenario where you are looking to run the command on a deployed application, instead of locally.
